docroot/modules/humsci/hs_dashboard/src/AnnouncementsManager.php

```diff
@@ -0,0 +1,291 @@
+          }
+
+          if (isset($data[1])) {
+            $data[1] = $this->convertDateToTimestamp(trim($data[1]));
+          }
+
+          if (isset($data[3])) {
+            $data[3] = $this->convertMarkdownLinks(trim($data[3]));
+          }
+
+          $rows[] = $data;
+        }
+        fclose($handle);
+      }
+    }
+
+    // Sort by date descending.
+    usort($rows, function ($a, $b) {
+      return $b[1] <=> $a[1];
+    });
+
+    // Convert dates.
+    foreach ($rows as &$row) {
+      $row[1] = $this->formatDate($row[1]);
+    }
+
+    return $rows;
+  }
+
+  /**
+   * Converts dates from Google Sheets into formatted dates.
+   *
+   * @param string $value
+   *   A string of text to convert into a formatted date.
+   *
+   * @return int
+   *   A Unix timestamp.
+   */
+  private function convertDateToTimestamp(string $value): int {
+    $value = str_replace("\u{A0}", ' ', $value);
+    $date = \DateTime::createFromFormat('M d, Y', $value);
+
+    return $date ? $date->getTimestamp() : 0;
+
+  }
+
+  /**
+   * Converts a Unix timestamp into a Drupal formatted date.
+   *
+   * @param int $timestamp
+   *   The Unix timestamp.
+   *
+   * @return string
+   *   A formatted Drupal date.
+   */
+  private function formatDate(int $timestamp): string {
+    return $this->dateFormatter->format($timestamp, 'medium');
+  }
+
+  /**
+   * Convert markdown links into HTML links.
+   *
+   * @param string $text
+   *   Text to covert from markdown into HTML.
+   */
+  private function convertMarkdownLinks(string $text): string {
+    $markdown_link_regex = "/\[(.*?)\]\((https?:\/\/.*?)\)/";
+
+    return preg_replace_callback($markdown_link_regex, function ($matches) {
+      $converted_link = Link::fromTextAndUrl(
+        $this->t('@link_text', ['@link_text' => $matches[1]]),
+        Url::fromUri($matches[2])
+      );
+
+      return $converted_link->toString()->__toString();
+    }, $text);
+
+  }
+
+  /**
+   * Returns table headers. These are statically set.
+   *
+   * @return array
+   *   An array of table headers.
+   */
+  public function getTableHeader(): array {
+
+    $tableHeader = [
+      [
+        'data' => $this->t('Date'),
+      ],
+      [
+        'data' => $this->t('Update'),
+      ],
+    ];
+
+    return $tableHeader;
+  }
+
+  /**
+   * Returns table rows based on the announcements found in csv.
+   *
+   * @return array
+   *   An array of table rows with announcement data.
+   */
+  public function getTableRows(): array {
+    $table_rows = [];
+
+    if ($cache = \Drupal::cache()->get('hs_dashboard_csv_announcements')) {
+      return $cache->data;
+    }
+    $csv_data = $this->getCsvAnnouncements(static::ANNOUNCEMENTS_CSV);
+
+    foreach ($csv_data as $row) {
+      $table_rows[] = [
+        'data' => [
+          ['data' => $row[1]],
+          ['data' => ['#markup' => $row[3]]],
+        ],
+      ];
+    }
+
+    \Drupal::cache()->set('hs_dashboard_csv_announcements', $table_rows, time() + 120);
```

Comment:
2 minutes seems a bit high. Maybe 5-10? I'm not sure what would be optimal here.